src/emulator/types.ts

```diff
@@ -1,21 +1,40 @@
 import { ChildProcess } from "child_process";
 import { EventEmitter } from "events";
+import * as path from "path";
 
 export enum Emulators {
   FUNCTIONS = "functions",
   FIRESTORE = "firestore",
   DATABASE = "database",
   HOSTING = "hosting",
+  PUBSUB = "pubsub",
 }
 
+export type JavaEmulators = Emulators.FIRESTORE | Emulators.DATABASE | Emulators.PUBSUB;
+
 // TODO: Is there a way we can just allow iteration over the enum?
 export const ALL_EMULATORS = [
   Emulators.FUNCTIONS,
   Emulators.FIRESTORE,
   Emulators.DATABASE,
   Emulators.HOSTING,
+  Emulators.PUBSUB,
 ];
 
+export const JAVA_EMULATORS = [Emulators.FIRESTORE, Emulators.DATABASE, Emulators.PUBSUB];
+
+export function isJavaEmulator(value: string): value is JavaEmulators {
+  if (isEmulator(value)) {
```

Comment:
equivalent to
```
return isEmulator(value) && JAVA_EMULATORS.indexOf(value) >= 0;
```

also, do we need the first clause?

Reply:
Thanks.  And yeah we do, to make TS not complain.  It makes `value` definitely an `Emulators`